File: 剑指Offer/6.从尾到头打印链表.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, x):
#         self.val = x
#         self.next = None

# 递归实现
class Solution:
    def __init__(self):
        self.ans = []

    def reversePrintHelper(self, head: ListNode) -> None:
        if head == None:
            return
        self.reversePrintHelper(head.next);
        self.ans.append(head.val)

    def reversePrint(self, head: ListNode) -> List[int]:
        self.reversePrintHelper(head)
        return self.ans        

File: 剑指Offer/6.从尾到头打印链表.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
 */

// 递归实现
class Solution {
    vector<int> ans;
    void reversePringHelper(ListNode* head) {
        if (head == NULL) {
            return;
        }

        reversePringHelper(head->next);
        ans.emplace_back(head->val);
    }

public:
    vector<int> reversePrint(ListNode* head) {
        reversePringHelper(head);
        return ans;
    }
};